### DISCUSSION TITLE
Add operator for DeferredTextQueryExpression

### DISCUSSION THREAD

**Discussion by nNipsx-Sec** [SigmaHQ/pySigma#50] [Ideas]
In some case of sigma rules and SIEM product, have multiple field Regex or CIDR and SIEM must have operator to define query example with Qradar: 
`ELECT UTF8(payload) as search_payload from events where "Commandline"='test' AND INCIDR('fieldA 192.168.0.0/16', "fieldA") OR INCIDR('fieldC 192.168.0.0/24', "fieldC")`

But now pysigma can't allow user define expression for that user just define 
```
    deferred_start : ClassVar[str] = ""
    deferred_separator : ClassVar[str] = ""
    deferred_only_query : ClassVar[str] = ""
```

I think that enough for prepare query. 
So my solution it's add `deferred_expression` or add more field operator for deferred example:
```
deferred_and
deferred_or
deferred_not
```

**Comment by thomaspatzke** [comment]
Unsure if I understood the issue, but the [`re_expression`](https://github.com/SigmaHQ/pySigma/blob/3c773bda98804dbe6013a319da317ed900b71704/sigma/conversion/base.py#L383) and [`cidr_expression`](https://github.com/SigmaHQ/pySigma/blob/3c773bda98804dbe6013a319da317ed900b71704/sigma/conversion/base.py#L389) properties of the `TextQueryBackend` should do what you need:

```
class CustomBackend(TextQueryBackend):
[...]
    cidr_expression = "INCIDR('{field} {value}', "{field}")
[...]
```

The purpose of deferred expressions is not to integrate them in the main condition vial boolean operators, but to defer them to a later stage of the whole query, after the main search condition. This was necessary in Splunk, because the main search conditions has no possibility to express regular expressions or CIDR as filters. This doesn't seems to be the case for QRadar and therefore the expressions can be defined as above without usage of deferred expressions.

**Reply by nNipsx-Sec** [reply]
Oh thanks for support. It's helpful for me
